zookeeper-server/src/test/java/org/apache/zookeeper/common/CertificatesToPlayWith.java

```diff
@@ -0,0 +1,569 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.zookeeper.common;
+
+/**
+ * Some X509 certificates to test against.
+ * <p>
+ * Note:  some of these certificates have Japanese Kanji in the "subjectAlt"
+ * field (UTF8).  Not sure how realistic that is since international characters
+ * in DNS names usually get translated into ASCII using "xn--" style DNS
+ * entries.  "xn--i8s592g.co.jp" is what FireFox actually uses when trying to
+ * find &#x82b1;&#x5b50;.co.jp.  So would the CN in the certificate contain
+ * "xn--i8s592g.co.jp" in ASCII, or "&#x82b1;&#x5b50;.co.jp" in UTF8?  (Both?)
+ * </p>
+ *
+ * @since 11-Dec-2006
```

Comment:
Do we have to cite the source of this code ?
Or did you create this file by yourself ?

If it is not from an ASF project we have to add some line in NOTICE file

Reply:
Tests and test resources (this file) have been copied from ASF project http client. Name of the project is Apache HttpComponents: https://hc.apache.org/
What do I need to add exactly?

Reply:
AFAIK (and Enrico's question also applies this) if it's from another Apache project, the Apache license header is enough.

Reply:
Just checked JVMPauseMonitor out of curiosity. I got that class from hadoop, but modified it. I do have a comment indicating the original source. Maybe adding something similar?
https://github.com/apache/zookeeper/blob/fe940cdd8fb23ba09684cefb73233d570f4a20fa/zookeeper-server/src/main/java/org/apache/zookeeper/server/util/JvmPauseMonitor.java

Reply:
I would just write something like
"These test case have been taken from the Apache HttpComponents project."
Nothing more.
It is just a test resource (but we are going to "release it", in the source release)

Reply:
Sure. Thanks.